src/os/CMakeLists.txt

```diff
@@ -114,17 +120,20 @@ if(HAVE_LIBAIO)
 endif()
 
 if(WITH_SPDK)
+  include(Builddpdk)
   include_directories("${CMAKE_SOURCE_DIR}/src/spdk/include")
   add_custom_target(build_spdk
     COMMAND
-    $(MAKE) DPDK_INC_DIR=${DPDK_INCLUDE_DIR}
-    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/spdk/lib
+    make DPDK_DIR=${CMAKE_SOURCE_DIR}/src/spdk/dpdk/build
+    DEPENDS build_dpdk
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/spdk
     COMMENT "spdk building")
+
   # TODO: should use add_library(spdk INTERFACE IMPORTED) instead in new cmake,
   # if INTERFACE is supported.
   foreach(lib nvme log env_dpdk util)
     add_library(spdk_${lib} STATIC IMPORTED)
-    add_dependencies(spdk_${lib} build_spdk)
+    add_dependencies(spdk_${lib} build_dpdk build_spdk)
```

Comment:
and please move this `foreach` block into `BuildDPDK.cmake`.

Reply:
also the target of `build_spdk` depends on `build_dpdk` by itself, why shall we add the dependency from `spdk_${lib}` to `build_spdk`?

Reply:
>>>>> "and please move this foreach block into BuildDPDK.cmake"
In my opinion, this foreach block is used to compile spdk library? is it right to move it to BuildDPDK.cmake? Please correct me, thanks.

Reply:
oic, leave it to a followup PR then.

Reply:
thanks.